options.h

```diff
@@ -308,6 +312,10 @@ struct pattern_options {
 	uint8_t verifier_feed_zone_is_default;
 	int32_t verifier_timeout;
 	uint8_t verifier_timeout_is_default;
+	uint8_t catalog_role;
+	uint8_t catalog_role_is_default;
+	const char* catalog_member_pattern;
+	const char* catalog_producer_zone;
```

Comment:
Back when the first PR got made, we kind-of decided catalog properties must be set on a zone. A reason for that being so that you cannot all of a sudden break things simply by re-applying patterns. Why should this be part of a pattern?

Reply:
I evaluated this extensively. Have a look at the catzones.tdir and the text in the man page to see how this all works. The re-applying patterns issues are also covered in catzones.tdir.
Short summary: A catalog producer (catalog_role == CATALOG_ROLE_PRODUCER) is only functional when in a zone clause which is checked by evaluating the value for "catalog_producer_zone" (which only makes sense in a pattern and not in a zone; i.e. catalog member zones are always instantiated through patterns).
Currently only a single catalog consumer is allowed, but this may change in the future. In that future extension there is no reason why you would not have a catalog member zone which is a catalog zone itself. We do it step by step. Currently it at least allows the user to configure a consumer zone in a for nsd conventional manner.